ArduCopter/mode.cpp

```diff
@@ -688,9 +688,19 @@ Copter::Mode::AltHoldModeState Copter::Mode::get_alt_hold_state(float target_cli
 // these are candidates for moving into the Mode base
 // class.
 
-float Copter::Mode::get_surface_tracking_climb_rate(int16_t target_rate, float current_alt_target, float dt)
+float Copter::Mode::get_surface_tracking_climb_rate(int16_t target_rate)
 {
-    return copter.get_surface_tracking_climb_rate(target_rate, current_alt_target, dt);
+    return copter.get_surface_tracking_climb_rate(target_rate);
+}
+
+bool Copter::Mode::get_surface_tracking_target_alt_cm(float &target_alt_cm) const
```

Comment:
This doesn't look correct.  These methods should either be directly on the Mode object (and other places call copter.flightmode->get_surface_tracking_target_alt_cm, or the mode should be calling `copter.get_surface_tracking_target_alt_cm(...)`.  We should not be adding shim functions as these were really only here to avoid code churn.

Reply:
A quick grep shows these are only used within the mode class, so we can move the implementation of all of these into the Mode namespace (the state structure being declared `static` in the base Mode object).

Reply:
In fact, they should be `Copter::Mode::SurfaceTracking::get_target_alt_cm` etc

Reply:
txs again for the review, the issue was that I didn't want to expand the size of the change by moving the surface-tracking features into the mode object and I also didn't want to split the features across the mode and upper vehicle code.  This is why I added in the shim function - to make it all very similar to the existing surface tracking setup.

Reply:
I've removed the existing surface tracking shim functions.. so that doesn't do everything requested but is perhaps better.  TBH, I think there are other bigger existing issues with the mode class in Copter, for example, why are the "init" "run", "is_autopilot", etc functions protected?  I feel like perhaps we should sort this out before moving more stuff into the Mode class..